ras-aer-handler.c

```diff
@@ -52,6 +57,86 @@ static const char *aer_uncor_errors[32] = {
 	[20] = "Unsupported Request",
 };
 
+#ifdef HAVE_AMP_NS_DECODE
+#define IPMITOOL_CMD "/usr/bin/ipmitool"
+#define DMIDECODE_CMD "/usr/sbin/dmidecode"
+static bool ampere_ipmitool = false;
+
+static void ras_report_aer_ipmi_init(void)
+{
+	struct utsname unm;
+	struct stat st;
+	int rc;
+
+	/*
+	 * Verify on startup if we are on an Ampere Altra or Altra Max
+	 * platform, to set the use of ipmitool (if installed).
+	 */
+	if (stat(IPMITOOL_CMD, &st) != 0)
+		return;
+
+	if ((uname(&unm) != 0) || (strncmp(unm.machine, "aarch64", 8) != 0))
+		return;
+
```

Comment:
I would change the order here, checking first if the CPU is compatible with this report.
If the CPU/DMI entry is `Ampere(R) Altra(R)`, if the checks for arm64 and for the `IPMITOOL_CMD` fail, it should either print an error or an warning.

Reply:
v2: This code has significantly changed as now calling the new 'is_ampere_altra()' function to identify if we are running on an appropriate platform.
The verification order is now the way you suggested, we first verify the platform and if we are on a platform needing ipmitool and if the configured IPMITOOL command is not installed a Warning message is given on the daemon startup.